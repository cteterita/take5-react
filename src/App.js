import React, { useState, useEffect } from 'react';
import { Route, Switch, useHistory } from 'react-router-dom';

import firebase from './firebase';
import AuthContext from './AuthContext';
import DayView from './DayView/DayView';
import LandingView from './LandingView/LandingView';
import UserInfo from './UserInfo/UserInfo';

function App() {
  const [currentUser, setCurrentUser] = useState();
  const history = useHistory();

  const auth = firebase.auth();
  const provider = new firebase.auth.GoogleAuthProvider();
  const signInWithGoogle = () => {
    auth.signInWithPopup(provider)
      .then(history.push('/today'));
  };
  const signInAnonymously = () => {
    auth.signInAnonymously()
      .then(history.push('/today'));
  };
  const upgradeAnonymousAccount = () => {
    auth.currentUser.linkWithPopup(provider)
      .then((usercred) => {
        const { user } = usercred;
        console.log('Anonymous account successfully upgraded', user); // TODO: user feedback
      }).catch((error) => {
        console.log('Error upgrading anonymous account', error);
      });
  };
  const signOut = () => {
    auth.signOut()
      .then(history.push('/'));
  };

  useEffect(() => {
    auth.onAuthStateChanged((updatedUserAuth) => {
      setCurrentUser(updatedUserAuth);
    });
  });

  const contextValue = {
    currentUser,
    signInWithGoogle,
    signInAnonymously,
    signOut,
    upgradeAnonymousAccount,
  };

  return (
    <AuthContext.Provider value={contextValue}>
      <main className="App">
        <header>
          <h1>Take 5</h1>
          <UserInfo />
        </header>
        { /* Routes Here */ }
        <Switch>
          <Route path="/today" component={DayView} />
          <Route path="/:date(\d{4}-\d{2}-\d{2})" component={DayView} />
          <Route path="/" component={LandingView} />
        </Switch>
        <footer>
          (c) 2020 Claire Teter Lesh
          <br />
          <a href="https://github.com/cteterita/take5-react" target="_blank" rel="noopener noreferrer">Github</a>
          <br />
          <a href="https://docs.google.com/forms/d/e/1FAIpQLScz_YgbHT38M95Zn3xu1AhHGNT93GXElDGiFQ2Qsehi3r0ygA/viewform?usp=sf_link" target="_blank" rel="noopener noreferrer">Feedback</a>
        </footer>
      </main>
    </AuthContext.Provider>
  );
}

export default App;
