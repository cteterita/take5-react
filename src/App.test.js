import React from 'react';
import { BrowserRouter } from 'react-router-dom';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import App from './App';

describe('App component', () => {
  it('renders without errors', () => {
    const div = document.createElement('div');
    ReactDOM.render(<BrowserRouter><App /></BrowserRouter>, div);
    ReactDOM.unmountComponentAtNode(div);
  });
  it('renders as expected', () => {
    const tree = renderer.create(<BrowserRouter><App /></BrowserRouter>).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
