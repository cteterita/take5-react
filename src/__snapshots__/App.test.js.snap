// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App component renders as expected 1`] = `
<main
  className="App"
>
  <header>
    <div
      className="user-info-holder"
    />
    <h1>
      Take 5
    </h1>
    <div
      className="user-info-holder"
    />
  </header>
  <section
    id="daily-entry"
  >
    <h3>
      Take 5 minutes each day for yourself
    </h3>
    <p>
      Take the first 2.5 minutes after waking up to set your intentions for the day.
    </p>
    <p>
      Return in the evening for 2.5 minutes to review your day and practice gratitude.
    </p>
    <div
      id="sign-in-holder"
    >
      <button
        onClick={[Function]}
        type="submit"
      >
        Sign in with Google
      </button>
       or 
      <button
        onClick={[Function]}
        type="submit"
      >
        Try it Anonymously
      </button>
    </div>
  </section>
  <footer>
    Copyright 2020, 
    <a
      href="https://github.com/cteterita/take5-react"
      rel="noopener noreferrer"
      target="_blank"
    >
      Claire Teter Lesh
    </a>
  </footer>
</main>
`;
